plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/EclipsePreferenceState.java

```diff
@@ -0,0 +1,70 @@
+package io.snyk.eclipse.plugin.properties.preferences;
+
+import java.util.List;
+
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+
+import com.google.gson.Gson;
+
+import io.snyk.eclipse.plugin.utils.SnykLogger;
+import io.snyk.languageserver.protocolextension.messageObjects.FolderConfig;
+
+public class EclipsePreferenceState {
```

Comment:
Let's name it for what it does: it handles folder-specific settings. So something like `FolderPreferences` or `FolderConfigs` would probably be appropriate. `EclipsePreferenceState` is too abstract to be easily understandable.